### PR TITLE
P0482R6 char8_t: A type for UTF-8 characters and strings

### LINKED ISSUE #2403: [2018-11 CWG Motion 11] P0482R6 char8_t: A type for UTF-8 characters and strings (Revision 6)
[P0482R6](http://wg21.link/p0482r6)

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #2403.

### CODE REVIEW

**Comment by jensmaurer** [comment]
Notes:
P1073R3 Immediate functions (pull request #2461) contains a reorganization of the keyword table in [lex.key] as a \multicolfloattable, making further updates easier. It is recommended to apply the other pull request first and then rebase this one.

The present paper is missing an update for [atomics.alias], adding a typedef atomic_char8_t.

File: source/strings.tex [hunk 1]
```diff
@@ -6402,3 +6488,89 @@
...
+\returns
+The number of bytes stored in the array object (including any shift sequences).
+If \tcode{c8} does not contribute to a sequence of \tcode{char8_t}
+corresponding to a valid multibyte character,
+the value of the macro \tcode{EILSEQ} is stored in \tcode{errno},
+\tcode{(size_t) (−1)} is returned, and the conversion state is unspecified.
+
+\pnum
+\remarks
+Calls to \tcode{c8rtomb} with a null pointer argument for \tcode{buf}
```
Comment (cubbimew) [inline source/strings.tex]: There is no parameter named "buf". I think this "buf" should be "s" (same 2 lines lower).

Comment (jensmaurer) [reply source/strings.tex]: Right, fixed.

File: source/lex.tex [hunk 2]
```diff
@@ -728,6 +728,7 @@
 \tcode{static}            &
 \tcode{unsigned}          \\
 
+\tcode{char8_t}
```
Comment (Eelis) [inline source/lex.tex]: No & ?

Comment (jensmaurer) [reply source/lex.tex]: No. See the "notes" above:
"P1073R3 Immediate functions (pull request #2461) contains a reorganization of the keyword table in [lex.key] as a \multicolfloattable, making further updates easier. It is recommended to apply the other pull request first and then rebase this one."

**Comment by zygoloid** [comment]
> The paper is missing an update for [atomics.alias], adding a typedef atomic_char8_t. Assuming this is expressly not desired, leaving a fix to an LWG issue.

Perhaps I'm missing something, but I don't see why an update to [atomics.alias] would be warranted, since atomic_char8_t is not optional. The paper does update [atomics.syn] as necessary (that update was missing from the pull request; I've added it).

**Pull request merged**